src/middlewared/middlewared/plugins/kerberos.py

```diff
@@ -568,7 +570,7 @@ class Config:
             if entry['principal'] not in kerberos_principals:
                 kerberos_principals.append(entry['principal'])
 
-        return kerberos_principals
+        return kerberos_principals.sort()
```

Comment:
`sort()` method sorts in-place and returns `None`. You want `return sorted(kerberos_principals)`